### PR TITLE
Updated to support 2016-07-01-preview swagger spec

### PR DESCRIPTION
**Pull request by r22016**
- created swagger spec' to support latest API updates - https://raw.githubusercontent.com/r22016/azure-rest-api-specs/master/arm-servermanagement/2016-07-01-preview/servermanagement.json
- regenerated C#  implementation using AutoRest
- updated/added tests
- ran tests in 'record' and 'playback' mode - verified they pass

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@r22016__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (rameshr). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@r22016 You need to check in your rest api spec to the azure repo before submitting a PR here

**Comment by r22016** [comment]
the updated swagger spec was mereged last week - azure-rest-api-specs / arm-servermanagement / 2016-07-01-preview / servermanagement.json

**Comment by hovsepm** [comment]
@azuresdkci test this please

**Comment by hovsepm** [comment]
@azuresdkci test this please

File: src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/EncryptionJwkResource.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
```
Comment (cormacpayne) [inline src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/EncryptionJwkResource.cs]: @r22016 please use the latest version of the generator (0.17)

File: src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/RetentionPeriod.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
```
Comment (cormacpayne) [inline src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/RetentionPeriod.cs]: @r22016 same comment as above

File: src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/UpgradeMode.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
```
Comment (cormacpayne) [inline src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/UpgradeMode.cs]: @r22016 same comment as above

Comment (r22016) [inline src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Generated/Models/UpgradeMode.cs]: hi ,
Is this the right version ?

Microsoft (R) AutoRest 0.17.0.0

(after choco install autorest)

**Comment by cormacpayne** [comment]
@r22016 reverted the comments in the review since I made a mistake about the generator version you were using. You are fine using 0.16.0.0.

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@
 [assembly: AssemblyDescription("A convention-based library for saving and retrieving Azure Batch task output files.")]
 
 [assembly: AssemblyVersion("0.9.4.0")] 
-[assembly: AssemblyFileVersion("0.9.4.0")]
```
Comment (cormacpayne) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: @r22016 why was this change made?

Comment (r22016) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: I didn't make this change manually, could be the tooling, let me revert it.

**Comment by cormacpayne** [comment]
@r22016 since code was generated based on new Swagger, could you bump the minor version of this package?

**Comment by r22016** [comment]
updated package minor version
verified in build output: http://azuresdkci.cloudapp.net/job/sdk-for-net/3548/artifact/binaries/packages/ 
package version is now 1.0.8

**Comment by cormacpayne** [comment]
@r22016 you will need to also bump the version in AssemblyInfo:

https://github.com/r22016/azure-sdk-for-net/blob/66985771614982e209f0ac78954ed7dad69192ab/src/ResourceManagement/ServerManagement/Microsoft.Azure.Management.ServerManagement/Properties/AssemblyInfo.cs#L24

**Comment by r22016** [comment]
could you pl. review changes and let me know if you have anymore feedback

**Comment by cormacpayne** [comment]
LGTM

**Comment by r22016** [comment]
thanks, can we get the changes merged now?

**Pull request merged**